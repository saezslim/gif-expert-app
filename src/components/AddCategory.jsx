import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {
  const [inputValue, setInputValue] = useState("")
  
  const onInputChange = ({target}) => setInputValue(target.value)
  
  const onSubmit = (event) => {
    event.preventDefault()
    const originalValue = inputValue.trim()
    if(originalValue <= 3) return // Valida que el input al menos tenga 3 caracteres.
    onNewCategory(originalValue)
    setInputValue('')// Se limpia el input
  }
  return(
    <form onSubmit={onSubmit}>
      <input 
        type="text" 
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
        className="inputSearch"
      />
    </form>
  )
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func
}