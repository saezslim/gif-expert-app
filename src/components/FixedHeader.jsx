import { AddCategory } from "./"
import PropTypes from "prop-types"

export const FixedHeader = ({onNewCategory}) => {
  return(
    <header className="header">
      <h1 className="header__title">Gif Expert App</h1>
      <AddCategory onNewCategory={onNewCategory}/>
    </header>
  )
}

FixedHeader.propTypes = {
  onNewCategory: PropTypes.func
}