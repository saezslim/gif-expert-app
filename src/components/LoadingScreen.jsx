export const LoadingScreen = () => {
  return(
    <p>Cargando...</p>
  )
}