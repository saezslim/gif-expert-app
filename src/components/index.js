export * from './AddCategory'
export * from './GifGrid'
export * from './GifItem'
export * from './LoadingScreen'
export * from './FixedHeader'