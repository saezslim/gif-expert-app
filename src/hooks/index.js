export * from './useFetchGifs'
export * from './useFunCopy'