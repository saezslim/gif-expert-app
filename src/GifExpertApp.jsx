import { useState } from "react"
import { GifGrid, FixedHeader } from "./components"

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Dragon Ball'])
  
  const onAddCategory = (category) => {
    setCategories([category,...categories])
  }
  return(
    <>
      <FixedHeader
        onNewCategory={onAddCategory}
      />
      
        {
          categories.map((category) => (// Si le agregamos el id, renderiza por completo todo
            <GifGrid key={category}
             category={category} 
             />
          ))
        }
    </>
  )
}