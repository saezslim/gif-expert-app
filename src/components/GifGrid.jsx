import PropTypes from 'prop-types'
import {useFetchGifs} from "../hooks/useFetchGifs"
import { GifItem, LoadingScreen } from './'


export const GifGrid = ({ category }) => {
  const {images, isLoading} = useFetchGifs(category)

  return (
    <div className="grid-card">
      <h3 className="grid-card__title" aria-label='cardTitle'>{category}</h3>

      {
        !isLoading ?
        images.map((image) => (
          <GifItem key={image.id} {...image} />
        )) :
        <LoadingScreen />
    }
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
