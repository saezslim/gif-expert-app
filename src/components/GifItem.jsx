import { useFunCopy } from "../hooks";
import PropTypes from 'prop-types'

export const GifItem = ({ title, url }) => {
  const {isClicked, copyURL} = useFunCopy(url)
  return (
    <div className={isClicked ? 'card card--click' : 'card'} onClick={copyURL}>
      <h4 className="card__title">{title}</h4>
      <img className="card__img" src={url} alt={title} />
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
