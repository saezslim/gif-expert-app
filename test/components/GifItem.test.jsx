import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('Testing <GifItem />', () => {
  const title = 'Hola prueba'
  const url = 'http://urldepruebas.cl/'
  test('Debe hacer match con el snapshot.', () => {
    const {container} = render( <GifItem title={title} url={url} /> )
    expect(container).toMatchSnapshot()
  })

  test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />)
    const {src, alt} = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })
})