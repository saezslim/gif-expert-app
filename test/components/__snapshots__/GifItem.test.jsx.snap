// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem /> Debe hacer match con el snapshot. 1`] = `
<div>
  <div
    class="card"
  >
    <h4
      class="card__title"
    >
      Hola prueba
    </h4>
    <img
      alt="Hola prueba"
      class="card__img"
      src="http://urldepruebas.cl/"
    />
  </div>
</div>
`;
